src/lib.rs

```diff
@@ -353,6 +403,82 @@ impl Client {
         }
     }
 
+    /// Sets the given key.
+    ///
+    /// If `ttl` they will default to 0.  If the value is set
+    /// successfully, `Some(Value)` is returned, otherwise [`Error`] is returned.
+    /// NOTE: That the data is some Value is sparsely populated, containing only requested data by meta_flags
+    /// The meta set command a generic command for storing data to memcached. Based
+    /// on the flags supplied, it can replace all storage commands (see token M) as
+    /// well as adds new options.
+    //
+    // ms <key> <datalen> <flags>*\r\n
+    //
+    // - <key> means one key string.
+    // - <datalen> is the length of the payload data.
+    //
+    // - <flags> are a set of single character codes ended with a space or newline.
+    //   flags may have strings after the initial character.
+    pub async fn meta_set<K, V>(
+        &mut self,
+        key: K,
+        value: V,
+        ttl: Option<i64>,
+        meta_flags: FxHashMap<&[char], String>,
```

Comment:
Wondering if it makes more sense to take `meta_flags` input here as `&[char]`, same as we do for `meta_get`?  Ideally the parameter would be the same type across all meta methods to provide a cohesive experience for the user.

Alternatively, everything takes a HashMap of flags.  I understand the differentiation in implementations as-written today since currently we're only considering the flags that we need for probabilistic hotkey detection in `mg` which have no follow-on tokens, but when looking at the full spec we will need similar support between all meta methods.